src/main/java/org/toradocu/extractor/JavadocExtractor.java

```diff
@@ -1,515 +1,664 @@
-   *     documentation
+   * @param member an executable member
+   * @return the compilation unit where {@code member} is defined
    */
-  private List<ReturnTag> extractReturnTag(ExecutableMemberDoc member, ClassDoc classDoc)
-      throws IOException {
-    // List that will contain the return tags and will be returned by this method.
-    List<Tag> returnTags = new ArrayList<>();
-    final String TAG_NAME = "@return";
-
-    // Collect tags in the current method's documentation. This is needed because DocFinder.search
-    // does not load tags of a method when the method overrides a superclass' method also
-    // overwriting the Javadoc documentation.
-    Collections.addAll(returnTags, member.tags(TAG_NAME));
-
-    //    if (returnTags.isEmpty()) { // Inherit @return comments if necessary.
-    //      Doc holder = DocFinder.search(new DocFinder.Input(member)).holder;
-    //
-    //      // Collect tags that are automatically inherited (i.e., when there is no comment for a method
-    //      // overriding another one).
-    //      Collections.addAll(returnTags, holder.tags(TAG_NAME));
-    //
-    //      // Collect tags from method definitions in interfaces. This is not done by DocFinder.search
-    //      // (at least in the way we use it).
-    //      if (holder instanceof MethodDoc) {
-    //        ImplementedMethods implementedMethods =
-    //            new ImplementedMethods((MethodDoc) holder, configuration);
-    //        for (MethodDoc implementedMethod : implementedMethods.build()) {
-    //          Collections.addAll(returnTags, implementedMethod.tags(TAG_NAME));
-    //        }
-    //      }
-    //    }
-
-    List<ReturnTag> memberReturnTags = new ArrayList<>();
-    for (Tag tag : returnTags) {
-
-      // Handle inline taglets such as {@inheritDoc}.
-      TagletWriter tagletWriter =
-          new HtmlDocletWriter(configuration, DocPath.forClass(classDoc))
-              .getTagletWriterInstance(false);
-      String comment = tagletWriter.commentTagsToOutput(tag, tag.inlineTags()).toString();
-
-      /* taggedComment is the output of the Jsoup parsing: it can be exploited to extract
-       * text contained in a certain tag. For now we're interested in <code> */
-      Document taggedComment = Jsoup.parse(comment);
-
-      // Remove HTML tags (also generated by inline taglets). In the future, perhaps retain those
-      // tags, because they contain information that can be exploited.
-      comment = taggedComment.text();
-
-      ReturnTag tagToProcess = new ReturnTag(comment);
-      memberReturnTags.add(tagToProcess);
-    }
-
-    if (memberReturnTags.size() > 1) {
-      log.trace(
-          "There are more than one return tag in this method. This is not handled by Toradocu");
-    }
-
-    return memberReturnTags;
+  private CompilationUnit getCompilationUnit(CallableDeclaration<?> member) {
+    Optional<Node> nodeOpt = member.getParentNode();
+    Node node = nodeOpt.orElse(null);
```

Comment:
Under what circumstances can `nodeOpt` be absent?  Document.
If it isn't expected, use `orElseThrow`, or check `isPresent()` and throw an exception.